api/src/utils/get-ip-from-req.ts

```diff
@@ -1,16 +1,32 @@
 import { useEnv } from '@directus/env';
-import type { Request } from 'express';
+import type { IncomingMessage } from 'http';
 import { isIP } from 'net';
+import proxyAddr from 'proxy-addr';
 import { useLogger } from '../logger/index.js';
 
-export function getIPFromReq(req: Request): string | null {
+export function getIPFromReq(req: IncomingMessage): string | null {
 	const env = useEnv();
 	const logger = useLogger();
 
-	let ip = req.ip;
+	let trust: any = env['IP_TRUST_PROXY'];
+
+	// booleans must be passed as a function
+	if (typeof trust === 'boolean') {
+		trust = () => env['IP_TRUST_PROXY'];
```

Comment:
`IP_TRUST_PROXY` could be values other than `boolean`, we should handle those cases too. https://expressjs.com/en/guide/behind-proxies.html

Reply:
The `string` type was already handled by the `proxyAddr` library but number was not. I have updated it to account for all possible express values aside from function via [1918804](https://github.com/directus/directus/pull/25114/commits/191880431b3497e72621b8dc4615e244b07cba03)